src/tabmat/categorical_matrix.py

```diff
@@ -250,15 +255,43 @@ def __init__(
         cat_vec: Union[List, np.ndarray, pd.Categorical],
         drop_first: bool = False,
         dtype: np.dtype = np.float64,
+        cat_missing_method: str = "fail",
     ):
-        if pd.isnull(cat_vec).any():
-            raise ValueError("Categorical data can't have missing values.")
+        if cat_missing_method not in ["fail", "zero", "convert"]:
+            raise ValueError(
+                "cat_missing_method must be 'fail' 'zero' od 'convert', "
+                f" got {cat_missing_method}"
+            )
+        self._missing_method = cat_missing_method
+        self._missing_category = "(MISSING)"
 
         if isinstance(cat_vec, pd.Categorical):
             self.cat = cat_vec
         else:
             self.cat = pd.Categorical(cat_vec)
 
+        if pd.isnull(self.cat).any():
+            if self._missing_method == "fail":
+                raise ValueError(
+                    "Categorical data can't have missing values "
+                    "if cat_missing_method='fail'."
+                )
+
+            elif self._missing_method == "convert":
+                if self._missing_category not in self.cat.categories:
+                    self.cat = self.cat.add_categories([self._missing_category])
+                else:
```

Comment:
This is not really consistent with `pd.get_dummies`. If I have a category`(MISSING)` in a pandas series, then `pd.get_dummies` will not encode 1's for the NA's in the same columns as for category `(MISSING)`. 
```
x = pd.Series(["x", "y", "((MISSING)", pd.NA])
pd.get_dummies(x, dummy_na=True)
pd.get_dummies(x)
```
I think I would rather raise if `(MISSING)` is already a category...

Reply:
Good catch, thank you. I agree, raising would be a better/safer default.

Reply:
@MatthiasSchmidtblaicherQC, what do you think of the current solution? (raise if missing category exists, but also parametrize `_missing_category`)

Reply:
I like the solution. What error message does `self.cat = self.cat.add_categories([self._missing_category])` give if it fails because the missing category is already part of the categories? If it is unintuitive, we might want to raise an error with customised message in this case.

Reply:
It raises `ValueError: new categories must not include old categories: {'(MISSING)'}`, which I thought was okay, but I can of course replace it with a more tabmat-specific one.